lib/src/client.dart

```diff
@@ -121,6 +120,24 @@ class Client extends MatrixApi {
   /// The timeout until a typing indicator gets removed automatically.
   final Duration typingIndicatorTimeout;
 
+  DiscoveryInformation? _wellKnown;
```

Comment:
While I do think caching this is a good idea, where do we clear this? We probably should clear this on logout or in init even if the well-known request fails. And probably have a test for that.